### ISSUE TITLE
Possibility to disable extension for a specific branch

### ISSUE THREAD

**Issue by lmartineaumdf** [qoomon/maven-git-versioning-extension#46]
We`re having problem using this specific configuration in the master branch:

    <branch>
        <pattern>master</pattern>
        <versionFormat>${version}</versionFormat>
	 <property>
           <pattern>revision</pattern>
           <value>
               <pattern>.*</pattern>
                <format>${version}</format>
           </value>
         </property>         
     </branch>

We are using maven modules and we use dynamic variable in the module POM.xml, so we're using the flatten maven plugin. Long story story short, it seems to cause a conflict with the maven extension because maven is always complaining that it cannot find the ${revision} parent package

...
[ERROR] [ERROR] Some problems were encountered while processing the POMs:
[FATAL] Non-resolvable parent POM for com.company..common:commons-core:[unknown-version]: Failure to find com.company.common:common:pom:${revision} in ....

Maybe because the flatten plugin is merging the pom.xml AFTER the verison extension runs.

Anyway, I won't ask for help debugging this, but basically, what I would need, is to tell the extension to do... abolutely NOTHING in the branch master. I simply want to hard-code everything in the pom.xml once in the master branch. I've tried to remove the <branch> entry for master, I've also tried to use an empty configuration node.  With either configuration, the extensions falls into a default mode of using the commit hash as the version. Is there any way to tell the extension to do NOTHING ?  Don't touch the version number, don't touch the variable, I'll handle them.

For now, I'm removing the extension once in the master branch, but that's not ideal.

thanks

**Comment by qoomon** [comment]
I think I could add an maven parameter and/or and environment variable to disable the extension.
However could you provide a simple example project that have your described problems so i can debug it? would be great if I can find the bug and fix it.

**Comment by lmartineaumdf** [comment]
[example-project.zip](https://github.com/qoomon/maven-git-versioning-extension/files/3392737/example-project.zip)
see attached file. I've tried to simplified the project as much as could and kept only the relevant information.
As you can see, we have a <version> specified in the root pom.xml. This tag refers to a ${revision} property, defined in the same pom.xml. The maven module pom.xml inherits from this root pom.xml and also refers to the ${revision} property. At first, the ${version} from the extension didn't like the fact my version was referring to a property, so I've hard-coded the version directly in the version tag. But still didn't work for maven module that also refered to ${revision} property. 
Anyway, let me know if you have enough information with the attached file. Maybe it's the way we are configured with the flatten plugin..

**Comment by qoomon** [comment]
Found the problem in your project.
It is not allowed to use properties in `<version>` tag. Maven enforces you to use a static version string.

**Comment by qoomon** [comment]
However you can refer the other way around you can use `${project.version}` where ever you like

**Comment by lmartineaumdf** [comment]
So what would you propose considering our project structure with sub-module. Is this a bad idea to make maven module pom.xml inherit from the root pom ? How would you format that ? Would you be able to provide a revised version of the zip?
otherwise, could we go forward with a modification in the extension to make sure it doesn't touch anything in the master branch ?

**Comment by qoomon** [comment]
Can you clarify what you like to achieve?

**Comment by lmartineaumdf** [comment]
Consider the uploaded project structure (see previous post), I would like to make this config work:
<branch>
    <pattern>master</pattern>
    <versionFormat>${version}</versionFormat>
 <property>
       <pattern>revision</pattern>
       <value>
           <pattern>.*</pattern>
            <format>${version}</format>
       </value>
     </property>         
 </branch>

So basically, I want that, in my master branch, the extension to keep the final <version> tag generated by the pom.xml. But you're right, we are kinda illegal using a property in the <version> tag. But even if I hard-code the <version>  that would not solve the sub-module pom files where we want to refer to the parent pom revision. Maybe we're doing it bad though...

Basically, two scenario:
1- Could you provide us with an example of how to refactor our pom.xml (root and sub-module) to replicate the same structure (what we want is that all sub-mobule included in the repo share the same version number of the root pom.xml file)
2- Or could you add a mecanism to tell the extension that.. in the master branch, let the pom.xml resolve the version itself (even if it's a dynamic property).

**Comment by qoomon** [comment]
During research I recognised that I was not aware of this new `${revision}` feature https://maven.apache.org/maven-ci-friendly.html

I have fixed the a bug that actually causing your issue, I will release it today.
To achieve what you want just remove the `<property>` section within your master branch config, then it should work as expected

Thats how I've used version handling until today, have a look at my adjusted maven project 
[example-project.zip](https://github.com/qoomon/maven-git-versioning-extension/files/3400465/example-project.zip)
To change/update project version you should use the `mvn` comand `mvn versions:set -DnewVersion=<NEW_VERSION>` it will update all `pom.xml` versions including parent versions

**Comment by qoomon** [comment]
just released version `4.4.3`

**Comment by qoomon** [comment]
please confirm the fix

**Comment by lmartineaumdf** [comment]
Good morning! I didn't do exhaustive tests with all my projects, but after some smoke tests with version 4.4.3, it seems to work fine! It actually generated the war and sub-module with the correct version! Thanks for the modification, really appreciated!
